app/forms/declaration/TotalNumberOfItems.scala

```diff
@@ -58,35 +77,50 @@ object TotalNumberOfItems extends DeclarationPage {
 
   val removeCommasFirst = (validator: String => Boolean) => (input: String) => validator(input.replaceAll(",", ""))
   val notJustCommas = (input: String) => !input.forall(_.equals(','))
+  val removeCommasFirstOption = (validator: String => Boolean) => (input: Option[String]) => input.fold(false)(x => validator(x.replaceAll(",", "")))
+  val notJustCommasOption = (input: Option[String]) => !input.fold(false)(_.forall(_.equals(',')))
+  val wholeNumber = (validator: String => Boolean) =>
+    (input: String) =>
+      validator {
+        val decimalPoint = input.indexOf(".")
 
-  val equalsIgnoreCaseOptionString = (value: String) => (input: Option[String]) => input.map(_.equalsIgnoreCase(value)).getOrElse(false)
+        if (decimalPoint > 0) input.substring(0, decimalPoint)
+        else if (decimalPoint == 0) "0"
+        else input
+  }
+
+  val equalsIgnoreCaseOptionString = (value: String) => (input: Option[String]) => input.exists(_.equalsIgnoreCase(value))
   val isEmptyOptionString = (input: Option[String]) => isEmpty(input.getOrElse(""))
+  val nonEmptyOptionString = (input: Option[String]) => nonEmpty(input.getOrElse(""))
   val isAlphabeticOptionString = (input: Option[String]) => isAlphabetic(input.getOrElse(""))
   val lengthInRangeOptionString = (min: Int) => (max: Int) => (input: Option[String]) => lengthInRange(min)(max)(input.getOrElse(""))
 
-  def isFieldEmpty(field: String): Condition = _.get(field).map(_.isEmpty()).getOrElse(true)
-  def isFieldNotEmpty(field: String): Condition = _.get(field).map(!_.isEmpty()).getOrElse(false)
+  def isFieldEmpty(field: String): Condition = _.get(field).forall(_.isEmpty())
+  def isFieldNotEmpty(field: String): Condition = _.get(field).exists(_.nonEmpty)
+  def isFieldIgnoreCaseString(field: String, value: String): Condition = _.get(field).exists(_.equalsIgnoreCase(value))
+
+  def isAmountLessThan(field: String): Condition =
+    _.get(field).fold(false) {
+      wholeNumber(removeCommasFirst(x => Try(x.toInt).isSuccess && isNumeric(x) && (x.nonEmpty && x.toInt < 100000)))
+    }
+  def isNumber(field: String): Condition = _.get(field).exists(isNumeric)
 
   //We allow the user to enter commas when specifying these optional numerical values but we strip out the commas with `removeCommasFirst` before validating
   //the number of digits. To prevent the validation from allowing an invalid value like ",,,," we also must use the `notJustCommas`
   //function to specifically guard against this.
-  val mapping = Forms.mapping(
-    exchangeRate -> optional(
-      text()
-        .verifying(rateFieldErrorKey, isEmpty or (notJustCommas and removeCommasFirst(ofPattern(exchangeRatePattern))))
-    ),
-    totalAmountInvoiced -> optional(
+  val mapping: Mapping[TotalNumberOfItems] = Forms.mapping(
```

Comment:
This mapping is getting more and more complicated. Might be worth to split it in multiple parts (defs/vals)? Of course if feasible without a major effort